packages/marked-mathjax/src/index.ts

```diff
@@ -0,0 +1,74 @@
+import type { Marked } from 'marked';
+import type { MathJaxObject } from 'mathjax-full/js/components/startup.js';
+
+const startMath = /(\$|\\\(|\\\[)/;
+
+/**
+ * Adds an extension to marked to handle math in a manner compatible with our
+ * use of Mathjax in PrairieLearn.
+ *
+ * @param marked - The marked instance to extend.
+ * @param MathJax - The MathJax instance to use for rendering math. In
+ * client-side code, this is `window.MathJax`. In server-side code, this
+ * is the result of `require('mathjax-full').init()`.
+ */
+export function addMathjaxExtension(marked: Marked, MathJax: MathJaxObject) {
```

Comment:
Can we write any tests for this function? IIRC the way math is handled isn't trivial, so it'd be great to see some examples of what it should do in different scenarios (in addition to all the usual benefits of tests).

Reply:
This would require a bit of juggling to get a MathJax object passed as parameter. This is possible, but would probably require a limited version of server-side MathJax support (see other comment).

Reply:
I managed to do it, but MathJax does not support modules, so I had to work with a CJS require instead. I added a few tests, will continue with more tests tomorrow.

Reply:
Tests added in 5814e73831111cf024f5892bdb003bace106893d and 5a8ff142bb28ca4147d4e56fa83825540ed2140a.

Reply:
How are you running the tests? I'd expect to see a `test` script in `scripts` in the package's `package.json`; see most other packages for examples of this. You'd also want `mocha`, `tsx`, etc. as dev dependencies.

Reply:
Ah, I was running mocha directly in the test file. I'll add a test script.

Reply:
This is fixed in dabe61c628b4a6cec936e31b6f9dd59855bd265c and 13fa6d2178aefd1d7bf1cb863aa5cc031f45c59c.